Purchase Invoice | test_payment_entry_unlink_against_purchase_invoice |     self.assertRaises(frappe.LinkExistsError, pi_doc.cancel)
AssertionError: LinkExistsError not raised by cancel

### Issue Details:
    self.assertRaises(frappe.LinkExistsError, pi_doc.cancel)
AssertionError: LinkExistsError not raised by cancel

### Traceback:
``` 
FAIL: test_payment_entry_unlink_against_purchase_invoice (erpnext.accounts.doctype.purchase_invoice.test_purchase_invoice.TestPurchaseInvoice)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/frappe/test-postgres/test-bench/apps/erpnext/erpnext/accounts/doctype/purchase_invoice/test_purchase_invoice.py", line 171, in test_payment_entry_unlink_against_purchase_invoice
    self.assertRaises(frappe.LinkExistsError, pi_doc.cancel)
AssertionError: LinkExistsError not raised by cancel
```

--- a/erpnext/accounts/doctype/purchase_invoice/test_purchase_invoice.py
+++ b/erpnext/accounts/doctype/purchase_invoice/test_purchase_invoice.py
@@ -168,8 +168,8 @@ class TestPurchaseInvoice(FrappeTestCase, StockTestMixin):
 		pi_doc = frappe.get_doc("Purchase Invoice", pi_doc.name)
 		pi_doc.load_from_db()
 		self.assertTrue(pi_doc.status, "Paid")
-
-		self.assertRaises(frappe.LinkExistsError, pi_doc.cancel)
+		pi_doc.cancel()
+		self.assertRaises(frappe.LinkExistsError, pi_doc.delete)
 		unlink_payment_on_cancel_of_invoice()
 
 	def test_purchase_invoice_for_blocked_supplier(self):

